plugin/src/main/java/com/denizenscript/denizen/events/block/CauldronLevelChangeScriptEvent.java

```diff
@@ -69,28 +97,16 @@ else if (!changeType.equals("changes")) {
         return super.matches(path);
     }
 
-    @Override
-    public boolean applyDetermination(ScriptPath path, ObjectTag determinationObj) {
-        if (determinationObj instanceof ElementTag element && element.isInt()) {
-            event.setNewLevel(element.asInt());
-        }
-        return super.applyDetermination(path, determinationObj);
-    }
-
     @Override
     public ObjectTag getContext(String name) {
-        switch (name) {
-            case "location": return location;
-            case "cause": return new ElementTag(event.getReason());
-            case "old_level": return new ElementTag(event.getOldLevel());
-            case "new_level": return new ElementTag(event.getNewLevel());
-            case "entity":
-                if (event.getEntity() != null) {
-                    return new EntityTag(event.getEntity()).getDenizenObject();
-                }
-                break;
-        }
-        return super.getContext(name);
+        return switch (name) {
+            case "location" -> location;
+            case "cause" -> new ElementTag(event.getReason());
+            case "old_level" -> new ElementTag(event.getBlock().getBlockData() instanceof Levelled levelled ? levelled.getLevel() : 0);

```

Comment:
Same here, need to store the old level otherwise it'd keep reading the current level from the world and the context would change as the block changes (in e.g. an `after` event).
Can probably just store the old `BlockData` and use it for both this and the determination? Or storing an `int` and a `Material` separately would be fine as well.